ingest/change.go

```diff
@@ -82,6 +82,34 @@ func (c *Change) LedgerEntryChangeType() xdr.LedgerEntryChangeType {
 	}
 }
 
+// getLiquidityPool gets the most recent state of the LiquidityPool that exists or existed.
+func (c *Change) getLiquidityPool() (*xdr.LiquidityPoolEntry, error) {
+	var pre, post *xdr.LiquidityPoolEntry
+	if c.Pre != nil {
+		pre = c.Pre.Data.LiquidityPool
+	}
+	if c.Post != nil {
+		post = c.Post.Data.LiquidityPool
+	}
+	if pre == nil && post == nil {
+		return &xdr.LiquidityPoolEntry{}, errors.New(
+			"both pre and post entries cannot be nil; this change may not include a liquidity pool")
+	}
+	if post != nil {
+		return post, nil
+	}
+	return pre, nil
```

Comment:
If post is present, we always return it over pre, right? We don't need to track pre and post separately, then. Something like this would simplify the conditional logic:
```suggestion
	var entry *xdr.LiquidityPoolEntry
	if c.Pre != nil {
		entry = c.Pre.Data.LiquidityPool
	}
	if c.Post != nil {
		entry = c.Post.Data.LiquidityPool
	}
	if entry == nil {
		return &xdr.LiquidityPoolEntry{}, errors.New("this change does not include a liquidity pool")
	}
	return entry, nil
```

Reply:
I like it, it's cleaner, and tests still pass.